docker-storage-setup.sh

```diff
@@ -707,32 +707,56 @@ scan_disks() {
 }
 
 create_partition_sfdisk(){
-  local dev="$1" size
+  local dev="$1" size label="$2" type
   # Use a single partition of a whole device
   # TODO:
-  #   * Consider gpt, or unpartitioned volumes
+  #   * Consider unpartitioned volumes
   #   * Error handling when partition(s) already exist
   #   * Deal with loop/nbd device names. See growpart code
+
   size=$(( $( awk "\$4 ~ /"$( basename $dev )"/ { print \$3 }" /proc/partitions ) * 2 - 2048 ))
+  # if parted is not present, and this version of sfdisk
+  # does not support GPT then we are limited to 2TB
+  if [ $label = "gpt" ]; then
+    if $(sfdisk --help|grep gpt >/dev/null) ; then
+      label='label: gpt'
+      type=''
+      size=$(( $size - 34 ))
```

Comment:
Well, 34 is the first usable LBA sector when using GPT tables. `sfdisk` complains about not enough sectors available if we simply use the total number of sectors:

```
-bash-4.3# size=$(( $( awk "\$4 ~ /"$( basename /dev/sdb )"/ { print \$3 }" /proc/partitions ) * 2 - 2048 ))
-bash-4.3# echo $size
16775168
-bash-4.3# cat <<EOF | sfdisk /dev/sdb
> unit: sectors
> label: gpt
> 
> /dev/sdb1 : start=     2048, size=  ${size}
> EOF
Checking that no-one is using this disk right now ... OK

Disk /dev/sdb: 8 GiB, 8589934592 bytes, 16777216 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes

>>> Script header accepted.
>>> Script header accepted.
>>> Created a new GPT disklabel (GUID: 05700A90-D214-4CE9-BFFB-4251619AC4A5).
The last usable GPT sector is 16777182, but 16777215 is requested.
Failed to add partition: Invalid argument
Leaving.

-bash-4.3# size=$(( $size - 34 ))
-bash-4.3# echo $size
16775134
-bash-4.3# cat <<EOF | sfdisk /dev/sdb
unit: sectors
label: gpt

/dev/sdb1 : start=     2048, size=  ${size}
EOF
Checking that no-one is using this disk right now ... OK

Disk /dev/sdb: 8 GiB, 8589934592 bytes, 16777216 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes

>>> Script header accepted.
>>> Script header accepted.
>>> Created a new GPT disklabel (GUID: FC8D245F-4DA6-4E6D-9BED-BB17C4087FE8).
Created a new partition 1 of type 'Linux filesystem' and of size 8 GiB.
/dev/sdb2: 
New situation:

Device     Start      End  Sectors Size Type
/dev/sdb1   2048 16777181 16775134   8G Linux filesystem

The partition table has been altered.
Calling ioctl() to re-read partition table.
Syncing disks.
-bash-4.3# 
```

I don't know why this is not needed for MBR though.